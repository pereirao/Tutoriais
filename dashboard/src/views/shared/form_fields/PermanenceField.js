// @flow

import React, {PureComponent} from 'react';
import {Dropdown} from 'semantic-ui-react';
import I18n from '../../../localization';
import style_sheet from './form_fields.module.css';

const PERMANENCE_OPTIONS = ['temporary', 'permanent'];

type PermanenceFieldProps = { onChangeText: (string, Object) => void, options: Array<Object> };

class PermanenceField extends PureComponent {
  props: PermanenceFieldProps

  static defaultProps = {name: 'change_hours_permanence'}
  input: ReactClass<*>

  onChange = (_e, {value}) => {
    this.props.onChangeText(value, true);
  }

  render() {
    return (
      <Dropdown
        selection
        options={PERMANENCE_OPTIONS.map(value => ({
          value,
          text: I18n.t(`form.option.${value}`)
        }))}
        onChange={this.onChange}
        className={style_sheet.permanenceField}
        placeholder={I18n.t('form.placeholder.select_one')}/>
    );
  }
}

export default PermanenceField;
