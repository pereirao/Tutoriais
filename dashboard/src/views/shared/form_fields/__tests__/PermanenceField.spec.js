import React from 'react';

import PermanenceField from '../PermanenceField';

it('renders correctly', () => {
  expect(shallow(<PermanenceField />)).toMatchSnapshot();
});
