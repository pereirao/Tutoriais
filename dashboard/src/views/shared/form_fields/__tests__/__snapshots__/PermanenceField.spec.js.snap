// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Dropdown
  additionLabel="Add "
  additionPosition="top"
  className="permanenceField"
  closeOnBlur={true}
  icon="dropdown"
  minCharacters={1}
  noResultsMessage="No results found."
  onChange={[Function]}
  openOnFocus={true}
  options={
    Array [
      Object {
        "text": "Temporary",
        "value": "temporary",
      },
      Object {
        "text": "Permanent",
        "value": "permanent",
      },
    ]
  }
  placeholder="Select One..."
  renderLabel={[Function]}
  selectOnBlur={true}
  selection={true}
/>
`;
